pkg/auth/auth.go

```diff
@@ -237,3 +237,53 @@ func getAuthUrlBase() string {
 	}
 	return host
 }
+
+func getPem() (*rsa.PublicKey, error) {
+	writePemCache := false
+	// Try to read the PEM from local cache
+	configPath, err := cli_config.KlothoConfigPath(authServerPemCacheFile)
+	if err != nil {
+		return nil, err
+	}
+	bs, err := os.ReadFile(configPath)
+	// Couldn't read it from cache, so (a) try to fetch it from URL and (b) mark down that we should write it on success
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			zap.L().Debug("Couldn't read PEM cache file. Will download it.", zap.Error(err))
+		}
+		pemResp, err := http.Get(`https://klotho.us.auth0.com/pem`)
```

Comment:
this should likely be replaced with the method that can use env var overrides so we can test local and beta endpoints. This will otherwise expect our test auth envs to have the prod public key as the signing also